pkg/resmap/selector_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.
+
+package resmap_test
+
+import (
+	"testing"
+
+	"sigs.k8s.io/kustomize/v3/pkg/gvk"
+	"sigs.k8s.io/kustomize/v3/pkg/resmap"
+	"sigs.k8s.io/kustomize/v3/pkg/types"
+)
+
+func setupRMForPatchTargets(t *testing.T) resmap.ResMap {
+	result, err := rmF.NewResMapFromBytes([]byte(`
+apiVersion: group1/v1
+kind: Kind1
+metadata:
+  name: name1
+  labels:
+    app: name1
+  annotations:
+    foo: bar
+---
+apiVersion: group1/v1
+kind: Kind1
+metadata:
+  name: name2
+  labels:
+    app: name2
+  annotations:
+    foo: bar
+---
+apiVersion: group1/v1
+kind: Kind2
+metadata:
+  name: name3
+  labels:
+    app: name3
+  annotations:
+    bar: baz
+`))
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	return result
+}
+
+func TestFindPatchTargets(t *testing.T) {
+	rm := setupRMForPatchTargets(t)
+	testcases := []struct {
+		target types.Selector
+		count  int
+	}{
```

Comment:
please throw in a few tests with namespace "default" and "" in the selector matching against difference resources